### PR TITLE
MsApplicationPkg: Add FBPT Dump App

### PR DESCRIPTION
**Pull request by os-d**
## Description

Simple UEFI app that dumps FBPT into FBPT.bin. This previously lived in mu_basecore in 1c82e08470. However, it lived in an empty package and was not built for CI. Moving this to MsApplicationPkg makes more sense as it can be built in CI and used to verify Performance Data.

This is intended to enter mu_plus with the release/202405 integration, as release/202311 still carries the application.

- [ ] Impacts functionality?
  - **Functionality** - Does the change ultimately impact how firmware functions?
  - Examples: Add a new library, publish a new PPI, update an algorithm, ...
- [ ] Impacts security?
  - **Security** - Does the change have a direct security impact on an application,
    flow, or firmware?
  - Examples: Crypto algorithm change, buffer overflow fix, parameter
    validation improvement, ...
- [ ] Breaking change?
  - **Breaking change** - Will anyone consuming this change experience a break
    in build or boot behavior?
  - Examples: Add a new library class, move a module to a different repo, call
    a function in a new library class in a pre-existing module, ...
- [x] Includes tests?
  - **Tests** - Does the change include any explicit test code?
  - Examples: Unit tests, integration tests, robot tests, ...
- [ ] Includes documentation?
  - **Documentation** - Does the change contain explicit documentation additions
    outside direct code modifications (and comments)?
  - Examples: Update readme file, add feature readme file, link to documentation
    on an a separate Web page, ...

## How This Was Tested

From mu_basecore.

## Integration Instructions

Include the shell app in your DSC/FDF and run it to get the FBPT dumped out to a binary.

### CODE REVIEW

File: PerformancePkg/Application/FbptDump/FbptDump.inf [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  UefiApplicationEntryPoint
+  PrintLib
+  DebugLib
+  UefiBootServicesTableLib
+  UefiLib
+  SafeIntLib
+
+[Protocols]
+  gEfiAcpiSdtProtocolGuid
+  gEfiSmbiosProtocolGuid
```
Comment (makubacki) [inline PerformancePkg/Application/FbptDump/FbptDump.inf:38]: Please add an empty line to the end of this file.

File: PerformancePkg/Application/FbptDump/FbptDump.c [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+ This user-facing application dumps FBPT into a file.
+
+Copyright (c) Microsoft Corporation.
+SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#include <Uefi.h>
+#include <Library/BaseLib.h>
+#include <Library/BaseMemoryLib.h>
```
Comment (makubacki) [inline PerformancePkg/Application/FbptDump/FbptDump.c:11]: I noticed `BaseMemoryLib` and `MemoryAllocationLib` are not in the INF library classes section. Can you please double check library usage?

File: PerformancePkg/Application/FbptDump/FbptDump.c
Comment (makubacki) [inline PerformancePkg/Application/FbptDump/FbptDump.c:1]: Looks like the commit is still targeting `PerformancePkg`. It will also need to be added to the `MsApplicationPkg` build.

Comment (apop5) [reply PerformancePkg/Application/FbptDump/FbptDump.c:1]: it should target UefiTestingPkg, based from offline conversations.

Comment (makubacki) [reply PerformancePkg/Application/FbptDump/FbptDump.c:1]: Yeah, the decision was made for `UefiTestingPkg`. I still consider it more an "application" than a test, but those boundaries are debatable, and the testing package is fine.
